internal/provider/kubernetes/routes.go

```diff
@@ -110,6 +113,28 @@ func (r *gatewayAPIReconciler) processHTTPRoutes(ctx context.Context, gatewayNam
 					resourceMap.allAssociatedRefGrants[utils.NamespacedName(refGrant)] = refGrant
 				}
 			}
+
+			for i := range rule.Filters {
+				filter := rule.Filters[i]
+				if err := gatewayapi.ValidateHTTPRouteFilter(&filter); err != nil {
+					r.log.Error(err, "bypassing filter rule", "index", i)
+					continue
+				}
+
+				if filter.Type == gwapiv1b1.HTTPRouteFilterExtensionRef {
+					authenFilter, err := r.getAuthenticationFilter(ctx, httpRoute.Namespace, string(filter.ExtensionRef.Name))
```

Comment:
It might be so, that this call is triggered multiple times for the same authentication filter, as multiple httproutes could use the same authn filter.

If we just store authentication filter NamespacedName as key in resourceMap then after processing all httproutes we can do the `getAuthenticationFilter` call for unique filters.

Reply:
Commit [07e3398](https://github.com/envoyproxy/gateway/pull/813/commits/07e339888bc4aac8dfc1497d4dddca8b6902d934) resolves this comment. I took a slightly different approach than what you suggested, PTAL.